### PR TITLE
[WIP] Add RateLimiter in Scheduler

### PR DESCRIPTION
**Pull request by fabriziodemaria**
This is one of two proposed solutions for implementing throttling of the rate of creation of Docker containers from the scheduler.
The other path to be investigated aims at adding the throttler in the DockerOutputHandler instead.

### CODE REVIEW

File: styx-scheduler-service/src/test/java/com/spotify/styx/SchedulerTest.java [hunk 1]
```diff
@@ -324,6 +328,28 @@ public void shouldExecuteNewTriggers() throws Exception {
...
+    final RunState i0 = RunState.create(instance(WORKFLOW_ID1, "i0"), State.QUEUED, time);
+    init(i0);
+    scheduler.tick();
+
+    assertThat(stateManager.get(i0.workflowInstance()).state(), is(State.PREPARE));
+
+    final RunState i1 = RunState.create(instance(WORKFLOW_ID1, "i1"), State.QUEUED, time);
+    init(i1);
+    try {
+      await().atMost(Duration.ONE_SECOND).until(() -> scheduler.tick());
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/SchedulerTest.java]: we might need ```fail``` here to make sure the exception is thrown out.

File: styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java [hunk 2]
```diff
@@ -88,25 +89,38 @@
...
     this.ttls = Objects.requireNonNull(ttls);
     this.stateManager = Objects.requireNonNull(stateManager);
     this.workflowCache = Objects.requireNonNull(workflowCache);
     this.storage = Objects.requireNonNull(storage);
     this.triggerListener = Objects.requireNonNull(triggerListener);
+    this.rateLimiter = Objects.requireNonNull(rateLimiter);
   }
 
   void tick() {
+    try {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java]: a separated method maybe? ```tick()``` is getting longer. :)

File: styx-scheduler-service/src/test/java/com/spotify/styx/SchedulerTest.java (hunk 1 above)
Comment (fabriziodemaria) [inline styx-scheduler-service/src/test/java/com/spotify/styx/SchedulerTest.java]: Added

File: styx-common/src/main/java/com/spotify/styx/storage/Storage.java [hunk 3]
```diff
@@ -79,6 +79,11 @@
   boolean debugEnabled() throws IOException;
 
   /**
+   * Get the submission rate for Styx.
```
Comment (ulzha) [inline styx-common/src/main/java/com/spotify/styx/storage/Storage.java]: Not completely informative. It is "maximum rate" or something, not simply "rate". The documentation should describe units, too.

Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/storage/Storage.java]: "submission rate limit" ?

**Pull request merged**